core/testutils/src/main/java/com/mongodb/stitch/core/testutils/sync/ProxySyncMethods.kt

```diff
@@ -55,43 +58,89 @@ interface ProxySyncMethods {
-     * cache and if not found and there is internet connectivity, it is searched for remotely.
+     * Aggregates documents that have been synchronized from the remote
+     * according to the specified aggregation pipeline.
+     *
+     * @param pipeline the aggregation pipeline
+     * @return an iterable containing the result of the aggregation operation
+     */
+    fun aggregate(pipeline: List<Bson>): Iterable<Document?>
+
+    /**
+     * Counts the number of documents in the collection that have been synchronized from the remote
+     * according to the given options.
      *
-     * @param id the _id of the document to search for.
-     * @return the document if found locally or remotely.
+     * @param filter the query filter
+     * @return the number of documents in the collection
      */
-    fun findOneById(id: BsonValue): Document?
+    fun count(filter: Bson = BsonDocument()): Long
 
     /**
      * Updates a document by the given id. It is first searched for in the local synchronized cache
      * and if not found and there is internet connectivity, it is searched for remotely.
      *
-     * @param documentId the _id of the document to search for.
+     * @param filter the query filter
      * @param update the update specifier.
      * @return the result of the local or remote update.
      */
-    fun updateOneById(documentId: BsonValue, update: Bson): RemoteUpdateResult
+    fun updateOne(
+        filter: Bson,
+        update: Bson,
+        updateOptions: SyncUpdateOptions = SyncUpdateOptions()
+    ): SyncUpdateResult
+
+    /**
+     * Update all documents that have been synchronized from the remote
+     * in the collection according to the specified arguments.
+     *
+     * @param filter a document describing the query filter, which may not be null.
+     * @param update a document describing the update, which may not be null. The update to
+     * apply must include only update operators.
+     * @param updateOptions the options to apply to the update operation
+     * @return the result of the update many operation
+     */
+    fun updateMany(
+        filter: Bson,
+        update: Bson,
+        updateOptions: SyncUpdateOptions = SyncUpdateOptions()
+    ): SyncUpdateResult
 
     /**
      * Inserts a single document and begins to synchronize it.
      *
      * @param document the document to insert and synchronize.
      * @return the result of the insertion.
      */
-    fun insertOneAndSync(document: Document): RemoteInsertOneResult
+    fun insertOneAndSync(document: Document): SyncInsertOneResult
+
+    /**
+     * Inserts one or more documents. Begin synchronizing on the documents' ids.
+     *
+     * @param documents the documents to insert
+     * @return the result of the insert many operation
+     */
+    fun insertManyAndSync(documents: List<Document>): SyncInsertManyResult
 
     /**
      * Deletes a single document by the given id. It is first searched for in the local synchronized
```

Comment:
[nit] this comment looks out of date. honestly this file probably doesn't need doc comments for the CRUD methods, this is pretty internal